test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -2016,3 +2016,20 @@ def build(self):
     assert 'DEFINE conan_test_answer=456!' in c.out
     assert 'DEFINE conan_test_other=abc!' in c.out
     assert 'DEFINE conan_test_complex="1 2"!' in c.out
+
+
+def test_cxx_version_not_overriden_if_foced():
+    """Any C++ standard set in the CMakeLists.txt will have priority even if the
+    compiler.cppstd is set in the profile"""
+    tc = TestClient()
+    tc.run("new cmake_exe -dname=foo -dversion=1.0")
+
+    cml_contents = tc.load("CMakelists.txt")
+    cml_contents = cml_contents.replace("project(foo CXX)\n", "project(foo CXX)\nset(CMAKE_CXX_STANDARD 17)\n")
+    tc.save({"CMakeLists.txt": cml_contents})
+
+    # The compiler.cppstd will not override the CXX_STANDARD variable
+    tc.run("create . -s=compiler.cppstd=11")
+    assert "Conan toolchain: C++ Standard 11 with extensions OFF" in tc.out
+    # Conan says yes! But alas, the compiled code is C++17
+    assert "foo/1.0: __cplusplus2017" in tc.out
```

Comment:
THe ``__cplusplus`` is not universal, it is not the default for msvc

Reply:
Yup, suspected as much, adding the `/Zc:__cplusplus` is a bit noisy so I just disabled the configs